### DISCUSSION TITLE
Question: adding nsfs_account_config when using noobaa obc to create the claim in namespace type claim

### DISCUSSION THREAD

**Discussion by macthev** [noobaa/noobaa-core#7000] [General]
I have a namespace type resource called say xxxx backed bucketclass

run:
noobaa obc create obc-test2 --bucketclass xxxx

it creates the bucket nicely but what I can see is how to set the 'nsfs_account_config' on the auto account like when I create this manually. The noobaa cli obc generates the account name automatically but I can see how to call the noobaa cli to nfs_account_config details.

The api endpoint don't seem to have a way and update account won't let me.

my manual account with my added config.
nsfs_account_config:
gid: 1003
new_buckets_path: bucket-path/
nsfs_only: true
uid: 1003

**Comment by nimrod-becker** [comment]
The account is created automatically as part of the OBC, for any changes to the account you would need to update the account itself.

The are options in the CLI to update several things, if something specific is missing please let us know.

**Reply by macthev** [reply]
For nsfs, we need an nsfs_account_config to give the uid and gid for the filesystem. So we need a way for this to be assigned during the account create. For pvc obc type claims, the same somehow. Or allow one to specify and existing account that already has the attributes. I kinda like being able to predefined the account and use that one.

**Reply by macthev** [reply]
for obc, the account has a generated name so are you saying we should update that? ideally this could be specified as part  of the claim.

**Reply by liranmauda** [reply]
@macthev are you actually asking for an OBC with nsfs?
That is "add the option to create a Namespace FS OBC"? 

@nimrod-becker @dannyzaken WDYT?

**Reply by macthev** [reply]
@liranmauda  Yes asking for a way to use OBC with nsfs. 

When I create a nsfs store, I need to specify uid/gid in I think the nfs_config.  When creating with OBC,  Im using the noobaa cli and it generates an account with a generated name as part of the obc.  But for nsfs stores, it needs  the nfs_config(gid/uid) for the filesystem. But I can see no way to add after the fact as the name is generated.  Could be scripted I guess but would be nice is there was an option to create the account up front and then during the noobaa obc etc specify that account that already has the nfs_config(uid/gid).

This is so a container can make a claim and map to a known uid/gid in the filesystem,\. with out such mechanism, seems hard to use OBC with nsfs.

I hope I have explained better.
